packages/react-charts/src/components/ChartUtils/chart-patterns.tsx

```diff
@@ -328,13 +328,9 @@ export const getDefaultPatternProps = ({
   const defaultColorScale = getDefaultColorScale(colorScale, themeColorScale);
   let defaultPatternScale = patternScale;
   let isPatternDefs = !patternScale && hasPatterns !== undefined;
-  let patternId;
+  const patternId = React.useMemo(() => getPatternId(), []);
```

Comment:
Prior to this, `getPatternId` was only called conditionally. This changes it to being called every time, was the original author concerned about over calling and unique Ids becoming an issue?

Reply:
I assumed this change was necessary to solve the issue you're having?

I'd prefer that unique IDs are not created unless `isPatternDefs` is true -- no need to generate the ID, otherwise.

Reply:
I could add a conditional to the function in the `useMemo`, which would look like this:
```suggestion
  const patternId = React.useMemo(() => isPatternDefs ? getPatternId() : undefined, [isPatternDefs]);
```